### PR TITLE
Fixed task family insertion.

### PR DESCRIPTION
**Pull request by hjoliver**
Broken recently by #1493; looks like we don't have any tests for insertion of families (yet...)

### CODE REVIEW

File: lib/cylc/scheduler.py [hunk 1]
```diff
@@ -1307,35 +1307,32 @@ def command_dry_run_task(self, name, point_string):
...
-                matches.append(expr)
+                matching_tasks = [pattern]
             else:
                 # Regex task match.
-                for task in tasks:
-                    if re.match(expr, task):
-                        matches.append(task)
-        return matches
+                matching_tasks = [t for t in all_tasks if re.match(pattern, t)]
+        return matching_tasks
```
Comment (hjoliver) [inline lib/cylc/scheduler.py]: This function worked correctly, I've just improved it slightly and used clearer variable names.

File: lib/cylc/scheduler.py [hunk 2]
```diff
@@ -530,7 +530,7 @@ def command_insert_task(self, name, point_string, is_family,
                 submit_num = task_states_data[(task_name, task_point)].get(
                     "submit_num")
             new_task = self.config.get_task_proxy(
-                name, point, 'waiting', stop_point, submit_num=submit_num)
+                task_name, point, 'waiting', stop_point, submit_num=submit_num)
```
Comment (hjoliver) [inline lib/cylc/scheduler.py]: This is the bug.

**Comment by hjoliver** [comment]
Test added.  @matthewrmshin - please review.

**Comment by matthewrmshin** [comment]
@kaday please review 2.

**Comment by matthewrmshin** [comment]
Test OK in my environment.

**Comment by kaday** [comment]
Test battery passes and works in my environment.

**Pull request merged**